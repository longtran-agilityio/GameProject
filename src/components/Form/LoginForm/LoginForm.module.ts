// lib
import { styled, Typography, Box } from '@mui/material'

// theme
import { theme } from '@webapp/themes/index'

export const StyledForm = styled('form')(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  width: theme.spacing(50),
}))

export const StyledMessage = styled(Typography)(({ theme }) => ({
  color: theme.palette.error.main,
  fontStyle: 'italic',
  fontSize: theme.spacing(1.5),
  paddingBottom: theme.spacing(2),
}))

export const StyledBox = styled(Box)(({ theme }) => ({
  width: '90%',
  paddingBottom: theme.spacing(3),
  paddingTop: theme.spacing(0.1),
}))

export const StylesSubmitBtn = {
  padding: theme.spacing(1.2, 22),
  fontSize: theme.typography.h6.fontSize,
  marginTop: theme.spacing(5),
  color: theme.palette.background.paper,
  backgroundColor: theme.palette.primary.main,
  '&:hover': {
    color: theme.palette.text.secondary,
    backgroundColor: theme.palette.background.paper,
    boxShadow: 'none',
  },
  '& .Mui-focusVisible': {
    backgroundColor: theme.palette.background.paper,
  },
}
