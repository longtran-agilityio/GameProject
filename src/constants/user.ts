export enum User {
  USER = 'user',
  USER_CART = 'userCarts',
  LOGIN = 'login',
  REGISTER = 'register',
}
