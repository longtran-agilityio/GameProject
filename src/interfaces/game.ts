export interface IGame {
  id: number
  name: string
  prices: number
  released: string
  background_image: string
  description_raw: string
  genres: {
    name: string
  }[]
  platforms: {
    platform: {
      id: number
      slug: string
      name: string
    }
  }[]
  developers: {
    id: number
    name: string
    slug: string
  }[]
  publishers: {
    id: number
    name: string
  }[]
}

export interface ICartGame {
  userId: number
  cartGames: IGame[]
  id: number
}
