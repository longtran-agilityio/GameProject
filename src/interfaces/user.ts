export interface IUserLogin {
  email: string
  password: string
}

export interface IUserRegister extends IUserLogin {
  id: number
  firstName: string
  lastName: string
}

export interface ILoginForm {
  handleLogin: (data: IUserLogin) => void
  error: string
}

export interface IRegisterForm {
  handleRegister: (data: IUserRegister) => void
  error: string
}

export interface IUserAuthenticated extends IUserRegister {
  id: number
}

export interface IGetUser {
  accessToken: string
  user: IUserAuthenticated
}
