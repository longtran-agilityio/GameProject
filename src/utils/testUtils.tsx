import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from '@mui/material'
import { theme } from '@webapp/themes/index'

const AllBrowserRouter = ({ children }: { children: React.ReactNode }) => (
  <BrowserRouter>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </BrowserRouter>
)

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllBrowserRouter, ...options })

export * from '@testing-library/react'
export { customRender as render }
