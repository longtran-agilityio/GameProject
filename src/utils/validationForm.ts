export const EmailValidation = {
  required: 'Email is required',
  pattern: {
    value: /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/g,
    message: 'Invalid Email',
  },
}

export const PasswordValidation = {
  required: 'Password is required',
  minLength: { value: 8, message: 'min length is 8' },
}

export const FieldEmpty = {
  required: 'This field is required',
}
