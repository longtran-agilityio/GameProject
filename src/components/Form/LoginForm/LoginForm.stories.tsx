// libs
import { ThemeProvider } from '@mui/material'
import { ComponentStory, ComponentMeta } from '@storybook/react'

// component
import LoginForm from '.'

// theme
import { theme } from '@webapp/themes/index'

export default {
  title: 'Components/LoginForm',
  component: LoginForm,
} as ComponentMeta<typeof LoginForm>
const handleLogin = () => {}
const Template: ComponentStory<typeof LoginForm> = (args) => (
  <ThemeProvider theme={theme}>
    <LoginForm {...args} />
  </ThemeProvider>
)

export const LoginFormComponent = Template.bind({})
LoginFormComponent.args = {
  handleLogin: handleLogin,
}
