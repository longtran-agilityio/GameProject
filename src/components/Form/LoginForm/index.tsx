// lib
import { useForm, Controller } from 'react-hook-form'

// components
import { StyledInputBase } from '@webapp/components/Input/Input.module'
import { StyledButton } from '@webapp/components/Button/Button.module'

// styles
import { StyledForm, StyledMessage, StylesSubmitBtn } from './LoginForm.module'

// interface
import { ILoginForm, ILoginInput } from '@webapp/interfaces/user'

// validation
import { EmailValidation, PasswordValidation } from '@webapp/utils/validationForm'

const LoginForm = ({ handleLogin, error }: ILoginForm) => {
  const { control, handleSubmit } = useForm<ILoginInput>({ mode: 'onBlur' })

  const userLogin = (data: ILoginInput) => handleLogin(data)

  return (
    <StyledForm onSubmit={handleSubmit(userLogin)}>
      <Controller
        render={({ field, fieldState }) => (
          <StyledInputBase
            autoComplete={'off'}
            type='email'
            helperText={<StyledMessage>{fieldState.error?.message}</StyledMessage>}
            placeholder='Input email'
            {...field}
          />
        )}
        rules={EmailValidation}
        control={control}
        name='email'
      />

      <Controller
        render={({ field, fieldState }) => (
          <StyledInputBase
            autoComplete={'off'}
            type='password'
            helperText={<StyledMessage>{fieldState.error?.message}</StyledMessage>}
            placeholder='Input password'
            {...field}
          />
        )}
        rules={PasswordValidation}
        control={control}
        name='password'
      ></Controller>
      <StyledButton type='submit' sx={StylesSubmitBtn} variant='outlined'>
        Login
      </StyledButton>
      {error && <StyledMessage>{error}</StyledMessage>}
    </StyledForm>
  )
}

export default LoginForm
