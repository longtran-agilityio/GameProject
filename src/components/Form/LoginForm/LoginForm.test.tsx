// libs
import { render } from '@webapp/utils/testUtils'
import { faker } from '@faker-js/faker'

// component
import LoginForm from './index'

// interface
import { IUserLogin } from '@webapp/interfaces/user'

const handleUserLogin = () => {
  const loginRequest: IUserLogin = {
    email: faker.internet.email(),
    password: faker.internet.password(),
  }
  return loginRequest
}
const error = faker.datatype.string()
const LoginFormComponent = () => render(<LoginForm handleLogin={handleUserLogin} error={error} />)

describe('LoginForm', () => {
  test('should matches snapshot', () => {
    const { container } = LoginFormComponent()
    expect(container).toMatchSnapshot()
  })
  test('should show error when user login failed', () => {
    const { getByText } = LoginFormComponent()
    expect(getByText(error)).toBeInTheDocument()
  })
})
