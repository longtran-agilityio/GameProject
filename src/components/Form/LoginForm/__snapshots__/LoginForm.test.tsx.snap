// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should matches snapshot 1`] = `
<div>
  <form
    class="css-1j7o29d"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-fvh51n-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1tpp6ry-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-describedby=":r0:-helper-text"
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          name="email"
          placeholder="Input email"
          type="email"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-18iipir-MuiFormHelperText-root"
        id=":r0:-helper-text"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-s4icm2-MuiTypography-root"
        />
      </p>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-fvh51n-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1tpp6ry-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-describedby=":r1:-helper-text"
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r1:"
          name="password"
          placeholder="Input password"
          type="password"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-18iipir-MuiFormHelperText-root"
        id=":r1:-helper-text"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-s4icm2-MuiTypography-root"
        />
      </p>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1hd09f6-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Login
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <p
      class="MuiTypography-root MuiTypography-body1 css-s4icm2-MuiTypography-root"
    >
      ZnCMSn=C&gt;M
    </p>
  </form>
</div>
`;
